ui/src/app/views/recent-work/component.spec.ts

```diff
@@ -0,0 +1,157 @@
+    expect(cardNames).toEqual(['mockFile4.ipynb', 'mockFile3.ipynb', 'mockFile2.ipynb']);
+  }));
+
+  // it should not render the component at all if user has no cache
+  it('should not render if no cache', fakeAsync(() => {
+    userMetricsSpy.getUserRecentResources.and.returnValue(Observable.of([]));
+    updateAndTick(fixture);
+    const recentWork = fixture.debugElement.queryAll(By.css('.recent-work'));
+    expect(recentWork.length).toEqual(0);
+  }));
+
+  // neither scroll indicator should show if cache < 4
+  it('should not render either scroll indicator if cache fewer than 4', fakeAsync(() => {
+    userMetricsSpy.getUserRecentResources.and.returnValue(Observable.of([]));
+    updateAndTick(fixture);
+    const scrolls = fixture.debugElement.queryAll(By.css('.scroll-indicator'));
+    expect(scrolls.length).toEqual(0);
+  }));
+
+  // right scroll should appear (but no left scroll) if cache > 3
+  it('should render scroll indicators correctly if cache greater than 3', fakeAsync(() => {
+    userMetricsSpy.getUserRecentResources.and.returnValue(Observable.of(stubRecentResources(4)));
+    updateAndTick(fixture);
+    const de = fixture.debugElement;
+    const leftScroll = de.queryAll(By.css('#left-scroll'));
+    const rightScroll = de.queryAll(By.css('#right-scroll'));
+    expect(leftScroll.length).toEqual(0);
+    expect(rightScroll.length).toEqual(1);
+  }));
+
+  // test that component scrolls correctly
+  //    moves down list on right scroll click
+  //    right scroll disappears and left appears
+  //    moves up list on left scroll click
+  //    left scroll disappears and right appears
+  it('should scroll correctly', fakeAsync(() => {
+    userMetricsSpy.getUserRecentResources.and.returnValue(Observable.of(stubRecentResources(4)));
+    updateAndTick(fixture);
+    const de = fixture.debugElement;
+    const rightScroll = de.query(By.css('#right-scroll'));
+    const leftScroll = de.query(By.css('#left-scroll'));
+    simulateClick(fixture, rightScroll);
+    fixture.detectChanges();
+    tick();
+    // should have scrolled right so should be FIRST 3 and NOT last
+    expect(de.queryAll(By.css('.name')).map((card) => card.nativeElement.innerText.trim()))
+      .toEqual(['mockFile3.ipynb', 'mockFile2.ipynb', 'mockFile1.ipynb']);
+    // right scroll should not be present and left present
+    expect(rightScroll).toBeNull();
+    expect(leftScroll).toBeDefined();
+    simulateClick(fixture, de.query(By.css('#left-scroll')));
+    tick();
+    fixture.detectChanges();
+    // all should be returned to orig state
+    expect(de.queryAll(By.css('.name')).map((card) => card.nativeElement.innerText.trim()))
+      .toEqual(['mockFile4.ipynb', 'mockFile3.ipynb', 'mockFile2.ipynb']);
+    expect(rightScroll).toBeDefined();
+    expect(leftScroll).toBeNull();
+  }));
+});
+
+function stubRecentResources(numberOfResources: number) {
+  const currentCache = [];
+  while (numberOfResources > 0) {
+    const currentResource = {
```

Comment:
nit: I would probably just inline this below, I don't think the local var is helping readability